### ISSUE TITLE
"xterm" terminal type isn't quite right

### ISSUE THREAD

**Issue by athornton** [jupyterlab/jupyterlab#3413]
The default value of TERM in a terminal tab is "xterm".  However, the arrow keys don't work correctly inside a vim session inside the terminal inside JupyterLab.

Now, maybe this is due to a broken termcap entry; the base distro I'm using is Centos 7.

If TERM is "xterm-color" or "ansi" the arrow keys work fine.  Since the terminal *is* color-capable, I suggest that the default simply be set to "xterm-color".

**Comment by blink1073** [comment]
Hi @athornton, [terminado](https://github.com/jupyter/terminado/blob/f13ac98cd2a7a4f21d82d461d341720885378f56/terminado/management.py#L148) is applying that term type in the absence of a previously set `TERM` environment variable.  Are you able to set that variable to `xterm-color` for users when launching their hub instances?

**Comment by rvalieris** [comment]
I tried to set this on the jupyterhub config:
`c.Spawner.environment = {
	'TERM': 'xterm-256color'
}`

the variable is set on a notebook kernel, but not on the terminal !

if I understand the code here right: https://github.com/jupyter/terminado/blob/master/terminado/management.py#L175

the TERM variable is always being overwritten by `self.term_settings.get("type", DEFAULT_TERM_TYPE)`, regardless of whether TERM was set before or not.

I think this could be fixed by first checking if `env['TERM']` is already set, and only overwrite if  term_settings["type"] is set, what do you think @blink1073  ?

I don't know if jupyter_server is setting term_settings['type'] somewhere, I couldn't find the relevant code.

**Comment by blink1073** [comment]
Hi @rvalieris, I'm afk for a long weekend, will circle back next week

**Comment by rvalieris** [comment]
no problem, have a nice weekend !

**Comment by blink1073** [comment]
> I think this could be fixed by first checking if env['TERM'] is already set, and only overwrite if term_settings["type"] is set, what do you think @blink1073 ?

That sounds right to me, mind making a PR to terminado?

**Comment by rvalieris** [comment]
>That sounds right to me, mind making a PR to terminado?

cool ! I will.

there is one thing I forgot to say tho, and it goes back to the original issue @athornton reported.
the TERM variable is supposed to communicate the capabilities of the terminal emulator, so the default value should be correctly set for everyone, it seems that terminado supports up to 256 colors: [script](https://gist.github.com/HaleTom/89ffe32783f89f403bba96bd7bcd1263/f08769c328002eaa86637761cbee621368181c7a)
![terminal](https://user-images.githubusercontent.com/6310472/132255504-02361fa4-a8c0-4e8d-8080-01f7eb55fa45.png)

terminado uses xterm.js to render the terminal, and [as far as I could find](https://github.com/xtermjs/xterm.js/pull/2198) "xterm-256color" seems to be the correct value.

so in addition to fixing the env override, I think we should also set the default to "xterm-256color"

**Comment by blink1073** [comment]
That sounds right to me too, also based on  https://wiki.archlinux.org/title/Xterm#TERM_Environmental_Variable

**Comment by rvalieris** [comment]
uh oh, I just realized if you start jupyterlab directly from the terminal the existing TERM variable will be used by terminado, this will cause issues if the TERM is too different...
I guess the original behavior of ignoring env[TERM] was right after all... sorry I didn't catch this earlier. I will make another PR to revert this part.

so that doesn't leave any easy way for the user to override the term, however the only reason I was trying to override it was to fix the "xterm" default value anyway which we already fixed, so I think its fine.
